### PR TITLE
i2176: Add metrics to barman

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-2176

The README should explain all.

### CODE REVIEW

File: backup/barman-montagu [hunk 1]
```diff
@@ -20,8 +20,10 @@ Options:
   --slot=<slot>             Replication slot to use on the db
   --volume_data=<volume>    Volume name or path for data [default: barman_data]
   --volume_logs=<volume>    Volume name or path for logs [default: barman_logs]
-  --volume_recover=<volume> Volume name or path for recovery [default: barman_recover]
-  --volume_nightly<volume>  Volume name or path for nightly recover [default: montagu_db_volume]
+  --volume_recover=<volume> Volume name or path for recovery
+                            [default: barman_recover]
+  --volume_nightly<volume>  Volume name or path for nightly recover
+                            [default: montagu_db_volume]
```
Comment (MartinEden) [inline backup/barman-montagu]: Turns out the long lines here were breaking docopt(!) - it wasn't parsing the default values.

File: backup/barman-montagu [hunk 2]
```diff
@@ -113,7 +115,10 @@ def start_barman(host, slot, password_group, tag, registry, pull,
         return barman
 
     pw = get_barman_passwords(password_group)
-    image_name = '{}/{}:{}'.format(registry, DOCKER_IMAGE, tag)
+
+    image_name = '{}:{}'.format(DOCKER_IMAGE, tag)
+    if registry:
+        image_name = '{}/{}'.format(registry, image_name)
```
Comment (MartinEden) [inline backup/barman-montagu]: So we can use locally tagged images in development.

File: backup/barman-montagu [hunk 3]
```diff
@@ -133,7 +142,8 @@ def start_barman(host, slot, password_group, tag, registry, pull,
     volumes = [volume_create(v, d) for v in named_volumes()]
     barman = d.containers.run(image_name, name=BARMAN_CONTAINER, detach=True,
                               volumes=volume_mappings, environment=environment,
-                              extra_hosts=extra_hosts, network=network)
+                              extra_hosts=extra_hosts, network=network,
+                              ports={"5000": 5000})
```
[the remark this answers is not in the document]
Comment (MartinEden) [reply backup/barman-montagu]: That's Flask's default port.

File: backup/README.md [hunk 4]
```diff
@@ -66,8 +66,37 @@ Or, for local testing you would want:
...
+
+```
+barman_running{database="montagu"} 1
+barman_ok{database="montagu"} 1
+barman_pg_version{database="montagu"} 10.3
+barman_available_backups{database="montagu"} 1
+barman_time_since_last_backup_seconds{database="montagu"} 54.008972
+barman_time_since_last_backup_minutes{database="montagu"} 0.9001495333333334
+barman_time_since_last_backup_hours{database="montagu"} 0.015002492222222222
+barman_time_since_last_backup_days{database="montagu"} 0.0006251038425925926
```
[the remark this answers is not in the document]
Comment (MartinEden) [reply backup/README.md]: Prometheus adds two labels to every metric: `instance_name` and `job`. Plus we have a manual label here: `database`. So if we have two barman instances tracking the same db (as we do: production and AWS) they will differ by `instance_name`. And if they are tracking different databases they will also differ by `database`.

**Comment by MartinEden** [comment]
Using the shared code now.

File: backup/barman-montagu (hunk 3 above)
[the remark this answers is not in the document]
Comment (littledice) [reply backup/barman-montagu]: Ah right. Got confused because I was using Flask behind nginx on port 80.

File: backup/Dockerfile [hunk 5]
```diff
@@ -8,7 +8,8 @@ RUN apt-get update && \
                 cron \
                 python3-pip \
                 python3-setuptools \
-                wget
+                wget \
+                python-flask
```
Comment (richfitz) [inline backup/Dockerfile]: can we [organise these alphabetically](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments) please?

File: backup/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (richfitz) [reply backup/README.md]: Can you copy this comment verbatim into the README and/or the app?

File: backup/barman-montagu (hunk 1 above)
Comment (richfitz) [reply backup/barman-montagu]: I didn't know you could break across multiple lines like that - that's really good to know!

**Pull request merged**